recipes/winpcap/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class WinPcapConan(ConanFile):
+    name = "winpcap"
+    description = "For many years, WinPcap has been recognized as the industry-standard tool for link-layer network access in Windows environments, " \
+                  "allowing applications to capture and transmit network packets bypassing the protocol stack, and including kernel-level packet filtering, " \
+                  "a network statistics engine and support for remote packet capture."
+    license = "BSD-3-Clause"
+    topics = ("conan", "packet", "capture", "filtering", "network", "protocol")
+    homepage = "https://www.winpcap.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "capture_type": ["bpf", "dag", "dlpi", "enet", "nit", "null", "libdlpi", "linux", "pf", "sita", "snit", "snoop", "win32"],
+        "remote": [True, False],
+        "bluetooth": [True, False],
+        "usb_sniffing": [True, False],
+        "usb_sniffing_device": "ANY",
+        "turbocap": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "capture_type": "null",
+        "remote": True,
+        "bluetooth": False,
+        "usb_sniffing": True,
+        "usb_sniffing_device": "/dev/usbmon",
+        "turbocap": False,
+    }
+
+    exports_sources = "CMakeLists.txt", "cmake/*", "patches/*"
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.turbocap = True
+            self.options.capture_type = "win32"
+        elif self.settings.os == "Linux":
+            self.options.capture_type = "linux"
+        if self.settings.os != "Linux":
+            del self.options.bluetooth
+            del self.options.usb_sniffing
+            del self.options.usb_sniffing_device
```

Comment:
@jgsogo 
c3i fails to build on MSVC@Windows.
See https://github.com/conan-io/conan-center-index/pull/6348#issuecomment-880414443
c3i says my recipe emits an `ConanInvalidConfiguration` when `os=Windows`.

The options inside `default_options` are indeed invalid on Windows,
but they are converted to valid options in `config_options`.